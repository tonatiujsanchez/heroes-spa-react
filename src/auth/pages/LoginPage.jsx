import { useContext } from "react"
import { useNavigate } from "react-router-dom"
import { AuthContext } from "../context"



export const LoginPage = () => {
    

    const { login } = useContext( AuthContext )
    const navigate = useNavigate()

    const handleLogin = () => {

        const lastPath = localStorage.getItem('lastPathHeroes') || '/'

        login('Brandon Hernandez')
        navigate(lastPath, { replace: true })
    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr />

            <button
                onClick={handleLogin}
                className="btn btn-primary"
            >
                Iniciar sessi√≥n
            </button>

        </div>
    )
}
