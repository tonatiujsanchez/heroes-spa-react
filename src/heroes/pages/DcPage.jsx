import { HeroList } from "../components"



export const DcPage = () => {
    return (
        <div>
            <h1>DC COMICS</h1>
            <hr />

            <HeroList publisher={'DC Comics'}/>
        </div>
    )
}
