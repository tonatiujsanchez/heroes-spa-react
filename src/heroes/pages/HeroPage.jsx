import { useMemo } from "react"
import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getHeroById } from "../helpers"

export const HeroPage = () => {

    
    const { id } = useParams()
    const navigate = useNavigate()

    const hero = useMemo( ()=> getHeroById(id), [id])

    const navigateBack = () => {
        navigate(-1)
    }


    if( !hero ) { return <Navigate to={'/marvel'} /> }

    return (
        <div className="row mt-5">
            <div className="col-4 animate__animated animate__bounceInLeft">
                <img 
                    src={`/assets/heroes/${ hero.id }.jpg`} 
                    alt={ hero.superhero }
                    title={ hero.superhero }
                    className="img-thumbnail"
                />
            </div>
            <div className="col-8">
                <h2>{ hero.superhero }</h2>
                <ul className="list-group list-group-flush">
                    <li className="list-group-item"><b>Alter ego:</b> { hero.alter_ego } </li>
                    <li className="list-group-item"><b>Publisher:</b> { hero.publisher } </li>
                    <li className="list-group-item"><b>First appearance:</b> { hero.first_appearance } </li>
                </ul>
                <h4 className="mt-3">Charecters</h4>
                <p>{ hero.characters }</p>

                <button onClick={ navigateBack } className="btn btn-outline-danger">
                    Regresar
                </button>
            </div>
        </div>
    )
}
