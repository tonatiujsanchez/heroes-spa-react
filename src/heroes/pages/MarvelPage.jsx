import { HeroList } from "../components"


export const MarvelPage = () => {
  return (
    <div>
      <h1>MARVEL COMICS</h1>
      <hr />

      <HeroList publisher={'Marvel Comics'} />
    </div>
  )
}
