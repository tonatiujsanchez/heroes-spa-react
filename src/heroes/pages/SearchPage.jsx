
import { useLocation, useNavigate } from "react-router-dom"
import queryString from 'query-string'

import { useForm } from "../../hooks"
import { HeroCard } from "../components"

import { getHeroesByName } from "../helpers"


const HeroesResult = ({ heroes, q }) => {

    return heroes.length === 0
        ? (
            <div className="alert alert-danger animate__animated animate__fadeIn">
                No hero with <b>{q}</b>
            </div>
        ):(
            heroes.map( hero => (
                <HeroCard key={hero.id} { ...hero } />
            ))
        )
}


export const SearchPage = () => {

    const navigate = useNavigate()
    const location = useLocation()

    const { q = '' } = queryString.parse(location.search) 

    const heroes = getHeroesByName(q)


    const { seachText, onInputChange } = useForm({
        seachText: q
    })


    const handleSearchSubmit = (event) => {
        
        event.preventDefault()

        navigate(`?q=${ seachText.trim() }`)
    }


    return (
        <>
            <h1 className="mt-5">SearchPage</h1>
            <hr />
            <div className="row">
                <div className="col-5">
                    <h4>Searching</h4>
                    <hr />
                    <form onSubmit={ handleSearchSubmit }>
                        <input 
                            type="text" 
                            name="seachText" 
                            id="seachText"
                            autoComplete="off"
                            placeholder="Search hero"
                            value={ seachText }
                            onChange={ onInputChange }
                            className="form-control mb-2"
                        />
                        <button className="btn btn-outline-primary">Search</button>
                    </form>
                </div>
                <div className="col-7">
                        <h4>Result</h4>
                        <hr />
                        {
                           q.trim() === ''
                                ?(
                                    <div className="alert alert-primary animate__animated animate__fadeIn">
                                        Search a hero
                                    </div>
                                ):(
                                    <HeroesResult heroes={ heroes } q={ q } />
                                )
                        }
                </div>
            </div>
        </>
    )
}
