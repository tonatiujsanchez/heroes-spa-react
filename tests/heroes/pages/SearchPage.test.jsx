import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/pages/SearchPage"

describe('Pruebas en SearchPage', () => {
    test('Debe de mostrarse correctamente con valores por defecto', () => {
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar a batman y el imput con el valor del queryString',()=> {
        
        const query = 'batman'
        const urlImg = '/assets/heroes/dc-batman.jpg'
        
        render(
            <MemoryRouter initialEntries={[`/search?q=${query}`]}>
                <SearchPage />
            </MemoryRouter>
        )
        
        const input = screen.getByRole('textbox')
        const img = screen.getByRole('img')

        expect( input.value ).toBe(query)
        expect( img.src ).toContain(urlImg)

        // Comprobar mediante estilos
        // const alerta = screen.getByLabelText('alert-danger')
        // expect( alerta.style.display ).toBe('none')
    })
})