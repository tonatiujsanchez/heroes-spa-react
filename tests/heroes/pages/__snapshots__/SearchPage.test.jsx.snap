// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchPage Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1
    class="mt-5"
  >
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control mb-2"
          id="seachText"
          name="seachText"
          placeholder="Search hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
